packages/expo-keep-awake/android/src/main/java/expo/modules/keepawake/ExpoKeepAwakeManager.java

```diff
@@ -0,0 +1,88 @@
+package expo.modules.keepawake;
+
+import android.app.Activity;
+import android.view.WindowManager;
+
+import org.unimodules.core.ModuleRegistry;
+import org.unimodules.core.Promise;
+import org.unimodules.core.errors.CurrentActivityNotFoundException;
+import org.unimodules.core.interfaces.ActivityProvider;
+import org.unimodules.core.interfaces.InternalModule;
+import org.unimodules.core.interfaces.ModuleRegistryConsumer;
+import org.unimodules.core.interfaces.services.KeepAwakeManager;
+
+import java.util.Collections;
+import java.util.List;
+
+public class ExpoKeepAwakeManager implements KeepAwakeManager, InternalModule, ModuleRegistryConsumer {
+
+  private final static String NO_ACTIVITY_ERROR_CODE = "NO_CURRENT_ACTIVITY";
+
+  private ModuleRegistry mModuleRegistry;
+  private boolean mIsActivated;
+
+  @Override
+  public void setModuleRegistry(ModuleRegistry moduleRegistry) {
+    this.mModuleRegistry = moduleRegistry;
+  }
+
+  private Activity getCurrentActivity() throws CurrentActivityNotFoundException {
+    ActivityProvider activityProvider = mModuleRegistry.getModule(ActivityProvider.class);
+    if (activityProvider.getCurrentActivity() != null) {
+      return activityProvider.getCurrentActivity();
+    } else {
+      throw new CurrentActivityNotFoundException();
+    }
+  }
+
+  @Override
+  public void activate(Promise promise) {
```

Comment:
Since this module should be used not only with a corresponding `Promise`, how about we move the promise rejection/resolution to `KeepAwakeModule`, and here just throw an exception? If you think we should, we could have two activates: `activate()` and `activate(Runnable/Function/Callback/Whatever)` that would be called inside the `Runnable` down below. What do you think?

Reply:
I like that, check if that's what you meant.